### PR TITLE
ci: add golangci-lint as a github action

### PR DESCRIPTION
**Pull request by sbinet**
Please take a look.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: golangci-lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
```
Comment (kortschak) [inline .github/workflows/golangci-lint.yml:7]: Turns out these lines are a bad idea. See gonum#1330.

Comment (sbinet) [reply .github/workflows/golangci-lint.yml:7]: ah, indeed, what I had in go-hep was:
```
on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]
  schedule:
    - cron: '0 2 * * 1-5'
```

**Comment by sbinet** [comment]
PTAL.

I've removed the "check only linting in diff _wrt_ `origin/master`" option.

**Comment by sbinet** [comment]
merging in (I just rebased _wrt_ what you'd approved)

**Pull request merged**